### PR TITLE
Create and hook up the units list component

### PR DESCRIPTION
**Pull request by hatched**
Display the units list in the inspector when the user clicks on the 'all units' action button.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2930/
Test FAILed.

File: jujugui/static/gui/src/app/components/inspector/test-inspector.js [hunk 1]
```diff
@@ -36,19 +36,44 @@ describe('Inspector', function() {
...
         return {name: 'demo'};
       }};
-    var getAppState = function() {
-        return 'inspector';
-      };
+    var appState = {
+      sectionA: {
+        metadata: {}
+      }};
+    var shallowRenderer = testUtils.createRenderer();
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector/test-inspector.js]: It looks like you could use `jsTestUtils.shallowRender()` here.

File: jujugui/static/gui/src/app/components/inspector/test-inspector.js [hunk 2]
```diff
@@ -73,4 +101,24 @@ describe('Inspector', function() {
...
+    var service = {
+      get: function() {
+        return 'demo';
+      }};
+    var appState = {
+      sectionA: {
+        metadata: {}
+      }};
+    var changeStub = sinon.stub();
+    var shallowRenderer = testUtils.createRenderer();
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector/test-inspector.js]: Again, could probably use your snazzy `jsTestUtils.shallowRender()` here.

File: jujugui/static/gui/src/app/components/inspector/test-inspector.js [hunk 3]
```diff
@@ -73,4 +101,24 @@ describe('Inspector', function() {
...
+    shallowRenderer.render(
+        <juju.components.Inspector
+          changeState={changeStub}
+          appState={appState}
+          service={service} />);
+    var output = shallowRenderer.getRenderOutput();
+    assert.equal(output.props.children[0].props.title, 'demo');
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector/test-inspector.js]: Did we decide against doing props compare in favour of element compare?

File: jujugui/static/gui/src/app/components/service-overview/service-overview.js [hunk 4]
```diff
@@ -31,6 +31,35 @@ YUI.add('service-overview', function() {
...
+    _navigate: function(e) {
+      var title = e.currentTarget.getAttribute('title');
+      var activeAction;
+      this.state.actions.some((action) => {
+          if (action.title === title) {
+            activeAction = action;
+            return true;
+          }
+      });
+      this.props.changeState(activeAction.state);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/service-overview/service-overview.js]: Nice.

File: jujugui/static/gui/src/app/components/service-overview/test-service-overview.js [hunk 5]
```diff
@@ -71,14 +71,50 @@ describe('ServiceOverview', function() {
...
         key="Units"
         title="Units"
         value={value}
-        icon={none}
-        action={none}
         valueType={none}
         link={none}
-        linkTitle={none} />);
+        linkTitle={none}
+        action={output.props.children[0].props.action} />);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/service-overview/test-service-overview.js]: We may need a comment here as this looks a little strange at first glance.

File: jujugui/static/gui/src/app/components/service-overview/test-service-overview.js [hunk 6]
```diff
@@ -71,14 +71,50 @@ describe('ServiceOverview', function() {
...
+    getStub.withArgs('units').returns({toArray: function() {
+      return [{}, {}];
+    }});
+    getStub.withArgs('id').returns('demo');
+    var service = {
+      get: getStub
+    };
+    var changeState = sinon.stub();
+    var shallowRenderer = testUtils.createRenderer();
+    stubIcons();
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/service-overview/test-service-overview.js]: I guess having stuff like this means we can't always use the shallowRender util.

File: jujugui/static/gui/src/app/utils/component-test-utils.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      https://github.com/facebook/react/blob/
+      dc2570e1ceebd2b4be7ebe0990f8524f6b53ea7c/src/test/ReactTestUtils.js#L347
+
+    @method shallowRender
+    @param {Object} component The components to render.
+    @param {Boolean} returnRenderer Whether or not it should return the
+      component instance or just the rendered output.
+    @return {Object} See returnRenderer parameter.
+  */
+  shallowRender: function(component, returnRenderer) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/component-test-utils.js]: Thanks!

File: jujugui/static/gui/src/app/utils/component-test-utils.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+var jsTestUtils = {
+
+  /**
+    shalloRender provides a convenience wrapper around the React
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/component-test-utils.js]: shallo(w)Render.

File: jujugui/static/gui/src/app/utils/component-test-utils.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    @return {Object} See returnRenderer parameter.
+  */
+  shallowRender: function(component, returnRenderer) {
+    var shallowRenderer = testUtils.createRenderer();
+    shallowRenderer.render(component);
+    return (returnRenderer) ?
+      shallowRenderer : shallowRenderer.getRenderOutput();
+  },
+
+  log: function(obj, showFn) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/component-test-utils.js]: Thanks!

**Comment by huwshimi** [comment]
Thanks @hatched this looks really good! :+1: with a few minor comments.

File: jujugui/static/gui/src/app/utils/component-test-utils.js (hunk 9 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/component-test-utils.js]: Also, some doc strings here might be nice.

File: jujugui/static/gui/src/app/components/service-overview/test-service-overview.js [hunk 10]
```diff
@@ -71,14 +71,50 @@ describe('ServiceOverview', function() {
...
-        icon={none}
-        action={none}
         valueType={none}
         link={none}
-        linkTitle={none} />);
+        linkTitle={none}
+        action={output.props.children[0].props.action} />);
+  });
+
+  it('clicks to All Units line item navigates to the unit list', function() {
```
Comment (makyo) [inline jujugui/static/gui/src/app/components/service-overview/test-service-overview.js]: These are usually read as a sentence

`it('navigates to the unit list when All Units is clicked', function() {....` maybe?

Comment (makyo) [inline jujugui/static/gui/src/app/components/service-overview/test-service-overview.js]: Mixing arrowfunc and function seems odd to me.

File: jujugui/static/gui/src/app/components/unit-list-item/test-unit-list-item.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+var juju = {components: {}};
+var testUtils = React.addons.TestUtils;
+
+describe('UnitListItem', () => {
```
Comment (makyo) [inline jujugui/static/gui/src/app/components/unit-list-item/test-unit-list-item.js]: As above, mixing arrowfunc and function seems weird

**Comment by makyo** [comment]
:+1: QA okay

What do you think about picking one - arrowfuncs or anonymous functions - for a style to use regularly in tests?

File: jujugui/static/gui/src/app/components/inspector/test-inspector.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector/test-inspector.js]: Yeah I actually wrote that wrapper while at the very end of this branch heh so a bunch weren't updated. I'll have to go back and update them.

File: jujugui/static/gui/src/app/components/inspector/test-inspector.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector/test-inspector.js]: We did, this must have just been an oversight. Fixed

File: jujugui/static/gui/src/app/components/service-overview/test-service-overview.js (hunk 10 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/service-overview/test-service-overview.js]: done

File: jujugui/static/gui/src/app/utils/component-test-utils.js (hunk 9 above)
Comment (hatched) [inline jujugui/static/gui/src/app/utils/component-test-utils.js]: done

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2931/
Test FAILed.

**Comment by hatched** [comment]
Thanks for the reviews!

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**